### DISCUSSION TITLE
Updating cache key without ever fetching the item

### DISCUSSION THREAD

**Discussion by pke** [vercel/swr#723] [Q&A]
Our search hook (useSWR) returns a bunch of results.
Would it be possible to update individual cache keys without those cache keys ever being used before?

```js
searchResult.forEach(item => mutate(`item/${item.id}`, item, false))
```

so that when the item is actually rendered the "smart" component can use it as:

```
const Item = ({ id }) => {
  const { item } = useSWR(`item/${id}`, get)
  return <...>
}
```

and the item is not fetched from the server again?

**Comment by sergiodxa** [comment]
Yes, that should work in the sense the component will use the cached data, however because how SWR works it will revalidate it when mounted so It will fetch it again. You can disable this with the `revalidateOnMount: false` option.

**Reply by pke** [reply]
And I would have to set this option in the `useSWR` call of the component I suppose?

**Reply by sergiodxa** [reply]
Yes, per call or using the global config provider
